### PR TITLE
onedrive: 2.4.11 -> 2.4.12

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

upstream update, https://github.com/abraunegg/onedrive/releases/tag/v2.4.12

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/sync/onedrive/default.nix [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+, fetchFromGitHub
+, autoreconfHook
+, ldc
+, installShellFiles
+, pkg-config
+, curl
+, sqlite
+, libnotify
+, withSystemd ? stdenv.isLinux
+, systemd ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/onedrive/default.nix:12]:
```suggestion
, systemd
```

**Comment by peterhoeg** [comment]
Cc @ianmjones

**Comment by ianmjones** [comment]
@peterhoeg unfortunately I no longer run this package, nor can I, and therefore can not review this PR.

Feel free to remove me from this package's maintainers list just now, or I can submit a PR to remove myself from the maintainers list when I get a chance.

**Comment by peterhoeg** [comment]
I cannot put into words how much I envy you not having to deal with onedrive......

I'll just replace you with myself.

Cc @srgom

**Pull request merged**